[IMP] Misc: Remove section from from-address

tiku-odoo opened the pull request:
The mail.force.smtp.from and mail.dynamic.smtp.from keys do not work in Odoo 15 and 16. This part should be merged with the From Filter section.

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/documentation/pull/3165).

tiku-odoo commented:
@Abridbus  @jqu-odoo When you have a moment can you review this PR? I've removed the section noted in the documentation task. https://www.odoo.com/web#id=3064153&cids=3&model=project.task&view_type=form

Do you want the first section edited or leave it as is? It seems a bit incomplete. 

> 
> Use a default “From” email address
> Sometimes, an email’s “From” (outgoing) address can belong to a different domain, and that can be a problem.
> 
> For example, if a customer with the email address mary@customer.example.com responds to a message, Odoo will try to redistribute that same email to the other subscribers in the thread. However, if the domain customer.example.com forbids that kind of usage for security, the email that Odoo is trying to redistribute would get rejected by some recipients’ email servers.
> 
> To avoid that problem, make sure all emails use a “From” address from the same authorized domain.

Abridbus reviewed (approved):
Wups, that's indeed something I missed in https://github.com/odoo/documentation/pull/1092

Thanks!

tiku-odoo commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -99,29 +99,6 @@ that Odoo is trying to redistribute would get rejected by some recipients' email
 
 To avoid that problem, make sure all emails use a "From" address from the same authorized domain.
```
```suggestion
To avoid this problem, Odoo has a notifications configuration. The following system parameters should be set in order for notifications to be sent out from Odoo: 

`mail.default.from_filter` : Value is a domain. This parameter is used to redistribute responses effectively; if a ‘From’ email address's domain doesn’t match this parameter, it is rewritten and [mail.default.from] @ [mail.catchall.domain] is used instead.

`mail.default.from` : Value is “notifications” by default. Only used when mail.default.from_filter does not match the author’s domain.

You access the **System Parameters** in :ref:`developer mode <developer-mode>` in the :menuselection:`Settings -->
Technical --> Parameters --> System Parameters` menu.
```

tiku-odoo replied:
@bouvyd  Hi Damien, when you have a moment can you review this doc? Thanks and happy holidays. 👍

Abridbus replied:
Hello @tiku-odoo 
I'm having a quick look, it's already explained in the email_domain file, should it be put together ? With a redirection for the other page ?

Abridbus replied:
Yes, already explained here:
content/applications/general/email_communication/email_domain.rst

tiku-odoo reviewed (commented):
@jqu-odoo @Abridbus @jevs-odoo 

When you have a moment can you take a look at this addition to the doc. Is this good to add?

tiku-odoo commented:
@Abridbus I've made those changes as you requested to lines 89-102. Thanks for the suggestion, I think this makes the document much more simpler to follow. 

@mivu-odoo When you have a moment can you review this change (lines 89-102). I removed a section that was outdated and linked to the default from address in the emain_domains doc.: 

_______________________________________________
If the MTA supports [SRS (Sender Rewriting Scheme)](https://en.wikipedia.org/wiki/Sender_Rewriting_Scheme), SRS can be enabled to handle these situations. However, that is more complex and requires more technical knowledge that is not meant to be covered by this documentation.

Instead, Odoo can be configured to do something similar by itself:

Set the Alias Domain name in the Settings ‣ General Settings ‣ Discuss.

Setting the domain alias configuration on Odoo.
Turn on [developer mode](https://www.odoo.com/documentation/16.0/applications/general/developer_mode.html).

Go to Settings ‣ Technical ‣ Parameters ‣ System Parameters.

Add one system parameter from the following list:

To use the same “From” address for all outgoing messages, use the key mail.force.smtp.from and set that address as value (such as outgoing@mycompany.example.com).

To keep the original “From” address for emails that use the same domain, but change it for emails that use a different domain, use the key mail.dynamic.smtp.from and set the value as the email address that should be used in those cases (such as outgoing@mycompany.example.com ).
_____________________________________________________________

tiku-odoo commented:
@Abridbus @jqu-odoo @alry-odoo 

I've reorganized the email_servers & email_domain docs. When you have some time can you review the changes? 

To recap: 

I've kept Outgoing and Incoming configurations in the email_servers doc and all email DNS in the email_domain doc. Hoping this will allow for better understanding and troubleshooting as the two docs split a lot of the information up prior. 

Thanks,

Tim

Abridbus commented on content/applications/general/email_communication/email_domain.rst:
```diff
@@ -1,37 +1,27 @@
-relays the reply to the followers as a notification.
+In order for servers to accept emails from Odoo on a more regular basis, one of the solutions is
+for customers to create rules within their own mailbox. A filter can be added to the email inbox so
+that when email is received from the catchall (catchall@yourdatabase.odoo.com) it is moved to the
```
There is no email coming from the catchall, 
As the standard (no configuration) for 15, 16 and following, email are received from 
' "name of the user connected" <notifications@dbname.odoo.com> '
Which would be translated by:
' "name of the user connected" <{ICP.mail.from.filter}@{mail.catchall.domain}>'
(where ICP stand for ir.config.parameters, which is the System Parameters as you might now it ;) )

Abridbus commented on content/applications/general/email_communication/email_domain.rst:
```diff
@@ -1,37 +1,27 @@
-customers, or vendors) are relayed on behalf of the database users. Messages sent back to the
-chatter from external users will appear in the chatter from their respective email addresses, or as
-they are listed in their Contacts record.
+If you notice a Odoo mail server on a blacklist notify Odoo via a new help ticket and the
+technical support team will work to get the servers removed from the blacklist.
```
Functional support team is doing it also :) Some ppl know how to handle it!
I would suggest:
```suggestion
If you notice a Odoo mail server on a blacklist notify Odoo via a new help ticket and the
support team will work to get the servers removed from the blacklist.
```

Abridbus commented on content/applications/general/email_communication/email_domain.rst:
```diff
@@ -1,37 +1,27 @@
-configured domain, rather than a generic email address domain (such as gmail.com, outlook.com,
-etc.).
+Should the Odoo database be using a custom domain for sending emails from Odoo there are three
+records that should be implemented on the custom domain's DNS to ensure the sending and receiving
```
SPF DKIM & DMARC are only impacting the deliverability, not the receiving part.

Also as we were again discussing today with @jad-odoo , unfortunately, there is use case where even with a full and correct DNS configuration, we are unable to guarantee that emails do not arrive in SPAM. As the final receiving mailbox configuration might decide something different.

That's the reason why we can see emails arriving in Spam in Outlook, even with DNS fully and correctly configured but everything working smoothly in Gmail.

Abridbus commented on content/applications/general/email_communication/email_domain.rst:
```diff
@@ -111,6 +99,11 @@ Conformance)` record of a domain name tell the destination server what to do wit
+.. example::
+   DMARC: TXT record
+
+   v=DMARC1; p=none; sp=none; aspf=r; adkim=r; pct=100; rua=mailto:example@company.com;
```
```suggestion
   v=DMARC1; p=none;
```

Let's simplify this at maximum.

This DMARC is enough for a first configuration (just won't do anything).
I would rather add a link to this page:
https://dmarc.org/overview/
Explaining which parameters can be added.

(This change is subject to discussion :)  -- maybe @jqu-odoo might have an opinion on it!)

tiku-odoo replied:
I've added this in as well, great suggestion!

Abridbus commented on content/applications/general/email_communication/email_domain.rst:
```diff
@@ -126,7 +119,14 @@ If the domain name uses :abbr:`DMARC (Domain-based Message Authentication, Repor
 Framework)` compliant or enable :abbr:`DKIM (DomainKeys Identified Mail)`.
 
-.. danger::
+If a partner, customer, or vendor, uses :abbr:`DMARC (Domain-based Message Authentication,
+Reporting, & Conformance)` and has defined one of these policies, the Odoo server cannot relay
+emails from this partner to the database users.
```
Not the case anymore with the notifications@ configuration.
As we do not spoof anymore

tiku-odoo replied:
Noted, I will remove this section.

Abridbus commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -6,14 +6,29 @@ Odoo Online or Odoo.sh users
+Documents in Odoo (such as a CRM opportunity, a sales order, an invoice, etc.) have a discussion
+thread, called *chatter*.
+
+When a database user posts a message in the chatter, this message is sent by email to the followers
```
sent by email to the followers of the record, except the sender (obviously)
-- dunno if worth to add tho

tiku-odoo replied:
I've added this in, thanks for pointing this out!

Abridbus commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -6,14 +6,29 @@ Odoo Online or Odoo.sh users
+of the document as a notification. If a follower replies to the message, the reply updates the
+chatter, and Odoo relays another reply to the followers as a notification. Messages sent back to
+the chatter from users or external users will appear in the chatter from their respective email, or
+asthe name listed in their Contacts record.
```
as the
```suggestion
as the name listed in their Contacts record.
```

Abridbus commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -22,16 +37,16 @@ The following sections below contain information on how to integrate an external
-   email server's reputation themselves.
+   If no one in the company is employed to manage email servers, Odoo Online and Odoo.sh are
+   strongly recommended. In those Odoo hosting types' email systems work instantly and are monitored
+   by professionals. Nevertheless, a company can use their own email server if they want to manage
+   the email server's reputation themselves.
```
This would imply to do the DKIM configuration according to their outgoing mail server ;)

tiku-odoo replied:
Yes and I've linked the doc for email DNS. Good call.

Abridbus commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -252,3 +166,68 @@ Now, go to :menuselection:`Email Marketing --> Settings` and enable :guilabel:`D
+
+Odoo relies on generic email aliases to fetch incoming messages.
+
+* **Reply messages** of messages sent from Odoo are routed to their original discussion thread (and
+  to the inbox of all its followers) by the catchall alias (**catchall@**).
```
```suggestion
* **Reply messages** of messages sent from Odoo are routed to their original discussion thread (and
  to the inbox of all its followers) by the alias of the model if there is any or by the catchall alias (**catchall@**).
```

Well, not sure it's really interesting to note this but at least it's more accurate ;)

tiku-odoo replied:
This is better, thanks for the edit! 👍

Abridbus commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -252,3 +166,68 @@ Now, go to :menuselection:`Email Marketing --> Settings` and enable :guilabel:`D
+
+* **Reply messages** of messages sent from Odoo are routed to their original discussion thread (and
+  to the inbox of all its followers) by the catchall alias (**catchall@**).
+* **Bounced messages** are routed to **bounce@** in order to track them in Odoo. This is especially
+  used in `Odoo Email Marketing <https://www.odoo.com/page/email-marketing>`__ to opt-out invalid
+  recipients.
```
No no,
The bounce address is used as a Return-Path (see: https://sendgrid.com/blog/what-is-return-path/) 

This is adding the red envelop on the record that has failed.
Big improvement is coming in master to be able to correctly parse the bounce reason instead of the current&ugly `No Error`

On the mass mailing side, the bounces are only opt-out based on this:
        

> If it bounced too much times (5)*
> in the last month and the bounced are at least separated by one week,
> to avoid blacklist someone because of a temporary mail server error,
> then the email is considered as invalid and is blacklisted.

\* : See _routing_handle_bounce in odoo/addons/mass_mailing/models/mail_thread.py

A log note is added on the contact under `Blacklisted Email Addresses` on the Email Marketing Configuration Menu

tiku-odoo replied:
This is super helpful information. I've added this to the doc. Thanks for sharing 💯

Abridbus commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -252,3 +166,68 @@ Now, go to :menuselection:`Email Marketing --> Settings` and enable :guilabel:`D
+configuration view.
+
+.. tip::
+   To edit catchall and bounce aliases, first activate the :ref:`developer mode <developer-mode>`.
+   Then, go to :menuselection:`Settings --> Technical --> Parameters --> System Parameters` to
+   customize the aliases (*mail.catchall.alias* & *mail.bounce.alias*).
```
tip but also be careful,
you might lose some emails if you do this, 
the change needs to be done prior to go live, once you go live, if customer reply to a catchall that been changed, well, the system will not recognise the local-part anymore.

tiku-odoo replied:
Good point, I've added this in.

tiku-odoo commented:
@Abridbus Thank you very much for the review. I've incorporated your changes and also added the changes for the CLI server configuration. When you have some time next week can you or @jqu-odoo review before I send this off to final review?

CLI add-ins: 

Lines 67-73
Lines 195-200
Lines 278-283

Thanks ahead of time for your help on these docs! 🥇 

CC: @jero-odoo (for peer review- note two docs email_servers / email_domain)

jero-odoo commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -252,3 +191,97 @@ Now, go to :menuselection:`Email Marketing --> Settings` and enable :guilabel:`D
+  alias (**catchall@**).
+* **Bounced messages** are used as a Return-Path. This is especially
+  used in `Odoo Email Marketing <https://www.odoo.com/page/email-marketing>`__. In this case bounces
+  are opt-out based on if the email bounced to many times (5) in the last month and the bounces are
```
> are opt-out based on if the email bounced to many times (5) in the last month and the bounces 
 too many times

jero-odoo replied:
```suggestion
  are opt-out based on if the email bounced too many times (5) in the last month and the bounces are
```

jero-odoo reviewed (commented):
Just the one change (line 214)

tiku-odoo commented:
@brse-odoo When you have a moment, can you review the two docs: email_servers.rst & email_domain.rst? Both have been reorganized for content to simplify for the user. 

(Files changed Tab > Load diff)

There is one small change on line 214 that Jess (jero) suggested that I will be making. On the Azure and Mailjet docs I simply added/changed anchors that I set in the two email docs.

Thanks in advance for your help reviewing these. 

=============================================================
CC: @jqu-odoo / @Abridbus  -- Can you take a look at these two docs by the end of the week?
For the CLI change I've added in Lines: 67-73, 195-200, 275-283

brse-odoo commented on content/applications/general/email_communication/email_domain.rst:
```diff
@@ -1,37 +1,34 @@
+SPAM Email
+----------
+
+It happens sometimes that some emails from Odoo are misclassified by the different email providers
```
```suggestion
Sometimes, emails from Odoo are misclassified by the different email providers
```

brse-odoo commented on content/applications/general/email_communication/email_domain.rst:
```diff
@@ -1,37 +1,34 @@
+
+It happens sometimes that some emails from Odoo are misclassified by the different email providers
+and end up in spam folders. At the moment, some settings are out of Odoo's control, notably the
+way the different email providers classify our emails according to their own restriction policy
```
```suggestion
way the different email providers classify Odoo's emails according to their own restriction policy
```

brse-odoo commented on content/applications/general/email_communication/email_domain.rst:
```diff
@@ -1,37 +1,34 @@
+inbox. It is also possible to add the Odoo database domain onto a safe senders list or whitelist
+on the receiving domain.
+
+If you notice a Odoo mail server on a blacklist notify Odoo via a new help ticket and the support
```
```suggestion
If an Odoo mail server appears on a blacklist, notify Odoo via a new help ticket and the support
```

brse-odoo commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -252,3 +191,97 @@ Now, go to :menuselection:`Email Marketing --> Settings` and enable :guilabel:`D
+  blacklisted. A log note is added on the contact under :guilabel:`Blacklisted Email Addresses` on
+  the :guilabel:`Email Marketing Configuration Menu`.
+
+  Messages that bounce in the chatter will populate a red envelop indicating the failed delivery,
```
```suggestion
  Messages that bounce in the chatter will populate a red envelope indicating the failed delivery,
```

brse-odoo commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -252,3 +191,97 @@ Now, go to :menuselection:`Email Marketing --> Settings` and enable :guilabel:`D
+To edit catchall and bounce aliases, first activate the :ref:`developer mode <developer-mode>`.
+Then, go to :menuselection:`Settings --> Technical --> Parameters --> System Parameters` to
+customize the aliases (*mail.catchall.alias* & *mail.bounce.alias*). These types of changes should
+be completed prior to going the database going live as if a customer replies after a change is
```
```suggestion
be completed prior to the database going live. If a customer replies after a change is
```

brse-odoo commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -252,3 +191,97 @@ Now, go to :menuselection:`Email Marketing --> Settings` and enable :guilabel:`D
+   :menuselection:`Settings --> Technical --> Automation --> Scheduled Actions` and look for
+   :guilabel:`Mail: Fetchmail Service`.
+
+Hosting a local email servers
```
```suggestion
Hosting local email servers
```

brse-odoo reviewed (commented):
Hi @tiku-odoo, I've finished my review of the two docs: email_domain.rst and email_servers.rst.

Great job on the content! The only changes I made were to grammar to improve/clarify certain sentences for the reader. Feel free to ping me if you have any questions about my suggestions!

tiku-odoo commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -252,3 +191,97 @@ Now, go to :menuselection:`Email Marketing --> Settings` and enable :guilabel:`D
+
+On-Premise databases can use a Local Server as an incoming email server. To do so create a new
+incoming email server by navigating to :menuselection:`Settings --> Technical --> Incoming`.
+Under :guilabel:`Server Type` select :guilabel:`Local Server` and under the
+:guilabel:`Configuration` a script will populate to execute in the
+:doc:`Command-line Interface <../../../developer/cli>`. This python script is unique to the
+database the email server is being configured in.
```
I've removed this for now. Thanks.

tiku-odoo commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -252,3 +191,97 @@ Now, go to :menuselection:`Email Marketing --> Settings` and enable :guilabel:`D
 server here selected for mass mails. Note that in this case, the domain's Sender Policy Framework
 (SPF) records must be set to include both transactional and mass mail servers.
+
+.. tip::
+   In SaaS and Odoo.SH Databases in order to utilize the pre-configured Odoo email server as a
+   transactional email server to supplement the mass email server navigate to
+   :menuselection:`Settings --> Technical --> Outgoing` and create a new outgoing email server. For
+   :guilabel:`Authenticate with` select :guilabel:`Command-line Interface`. Test the connection and
+   set a lower priority than the mass email server.
```
I've removed this section for now. This is targeting V16 only? @Abridbus

tiku-odoo commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -42,51 +60,31 @@ enable the :guilabel:`External Email Servers` option. Then, click :guilabel:`Sav
-An Office 365 server can be used if the database's hosting type is **Odoo on-premise**. Office 365
-SMTP relays are not compatible with Odoo Online or Odoo.sh unless Odoo is configured to
-:ref:`force the outgoing "From" address <email_communication/default_from>`.
+On-Premise databases can use the :ref:`Command-line Interface (CLI)
+<reference/cmdline/server/emails>` to add or edit outgoing email servers. This can be used to
+create a transactional email server.  For SMTP accounts using :abbr:`2FA (Two Factor
+Authentication)` an app password should be used. The server should also be set up in
+:menuselection:`Settings --> Custom Mail Servers --> Outgoing Mail Servers --> New`. Then the
+:guilabel:`Command Line Interface` option should be selected for the
+:guilabel:`Authentication method`.
```
I've removed this for now.

tiku-odoo commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -252,3 +191,97 @@ Now, go to :menuselection:`Email Marketing --> Settings` and enable :guilabel:`D
+  to the inbox of all its followers) by the alias of the model if there is any or by the catchall
+  alias (**catchall@**).
+* **Bounced messages** are used as a Return-Path. This is especially
+  used in `Odoo Email Marketing <https://www.odoo.com/page/email-marketing>`__. In this case bounces
```
I've modified this section as bit to fit better. Thanks for bringing this up :)

tiku-odoo commented on content/applications/general/email_communication/email_domain.rst:
```diff
@@ -111,6 +106,11 @@ Conformance)` record of a domain name tell the destination server what to do wit
 that fails the :abbr:`SPF (Sender Policy Framework)` and/or :abbr:`DKIM (DomainKeys Identified
 Mail)` check.
 
```
Great I've added this in. Thanks for clarifying.

bouvyd commented:
read diagonally, seems ok - consider i approve

jqu-odoo commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -99,100 +85,39 @@ message, Odoo will try to redistribute that same email to the other subscribers
-   Then, go to :menuselection:`Settings --> Technical --> Parameters --> System Parameters` to
-   customize the aliases (*mail.catchall.alias* & *mail.bounce.alias*).
+.. note::
+   The `mail.default.from_filter` works only for `odoo-bin` configurations, otherwise this
```
The odoo-bin configuration is not defined in the document, is this concept clear enough for the reader? While waiting for the CLI part to be added, I would use a more informative denomination such as "the default Odoo mail server".

tiku-odoo replied:
@jqu-odoo Thanks for pointing this out. I made a reference to the CLI doc which talks about the odoo_bin. Is that okay in your opinion? Otherwise I can change -- it is my understanding that the system parameter: `mail.default.from_filter` also works for custom email servers as well as the default Odoo mail server. Can you clarify a bit / make a suggestion to what it should be changed to? I agree there is some complexity to the original statement. Thanks :)

jqu-odoo replied:
Hey @tiku-odoo I can confirm that mail.default.from_filter works only with the default Odoo mail server, not with custom mail servers. Indeed you can call it odoo-bin if there is a link to the related documentation ;-)

tiku-odoo replied:
Can you check the configuration you left on this db: https://www.purpletest.net ? The mail.default.from_filter system parameter is present. I've changed the doc to reflect what you have noted. Thanks for pointing this out!

jqu-odoo replied:
The config parameter is there but it does nothing because there is an enabled mail server. I removed the mail.default.from_filter from your database to show that the from is still encapsulated.

jqu-odoo commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -231,14 +156,16 @@ first outgoing email server (sorted by priority).
-===================================================================
+-------------------------------------------------------------------
 
 In Odoo a separate Mail Transfer Agent (MTA) server can be used for transactional emails and mass
```
Mail Transfer Agent (MTA) is not used anywhere else. I would stick to the generic term "mail server" or "outgoing mail server".

tiku-odoo replied:
Noted, thanks.

tiku-odoo commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -6,14 +6,20 @@ Odoo Online or Odoo.sh users
 ============================
 
 Since **Odoo sets up its own mail servers for the database**, outgoing and incoming emails already
-work out-of-the-box. So for **Odoo Online** and **Odoo.sh** customers, nothing needs to be done!
```
Thanks for the suggestion. I have incorporated parts of this into the doc.

tiku-odoo commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -99,100 +85,39 @@ message, Odoo will try to redistribute that same email to the other subscribers
-      :align: center
-      :alt: System parameters with catchall configuration in Odoo.
+The field can be a domain name or an entire email address, or it can remain empty. If the sender’s
+email address does not match this set filter, then the email will be encapsulated using a
```
Leaving as is. Thanks for the suggestion!

tiku-odoo commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -252,3 +179,80 @@ Now, go to :menuselection:`Email Marketing --> Settings` and enable :guilabel:`D
+=======================
+
+Odoo relies on generic email aliases to fetch incoming messages.
+
```
Thanks for the suggestions, I have used some excerpts from this copy and incorporated into my draft.

tiku-odoo commented:
@StraubCreative This document is ready for your final / technical review. Thanks a head of time for your time and review 👍

samueljlieber commented:
Updates in https://github.com/odoo/documentation/commit/205bd4f076dfe7a7936c759ae1030624a3855366
- Fixed build errors
    - added a line at the end of `email_servers.rst`

@StraubCreative there still is a build error for a tab on line 142 🤔

samueljlieber commented:
Updates in  320065f 
- fixed build errors
   - removed tab from `email_domain.rst`
   - removed a file (`e -i HEAD~2.txt`) that was added accidentally during a squashing step

StraubCreative reviewed (changes requested):
Hi @tiku-odoo 
See change requests below.

Very small things on content itself, most of the issues involve RST:
- `pre` formatting for input values, such as DNS and email addresses (there are a lot of consistency issues here too, some are italicized, others are bold, some are pre...)
- line breaks

As well, some comments on the screenshots.
If the other images in `email_servers` are not being used for `email_servers.rst`, then let's remove them. If they're being used on another doc, then we'll have to move them.

Thanks and let me know if you have questions!

tiku-odoo commented on content/applications/general/email_communication/email_domain.rst:
```diff
@@ -1,37 +1,35 @@
+inbox. It is also possible to add the Odoo database domain onto a safe senders list or whitelist
+on the receiving domain.
+
+If an Odoo email server appears on a blacklist, notify Odoo via a new help ticket and the support
```
Yes, this is an excellent suggestion, thanks 💯

tiku-odoo commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -99,107 +93,47 @@ message, Odoo will try to redistribute that same email to the other subscribers
-   .. image:: email_servers/system-parameters.png
-      :align: center
-      :alt: System parameters with catchall configuration in Odoo.
+The field can be a domain name or an entire email address, or it can remain empty. If the sender’s
+email address does not match this set filter, then the email will be encapsulated using a
+combination of the two system parameters: mail.default.from and mail.catchall.domain. For example,
+“Admin” <admin@example.com> =>“Admin” <notifications@mycompany.com>.
```
Good call, I've elaborated a bit more on the example. Thanks for pointing this out.

tiku-odoo commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -249,6 +185,86 @@ to give priority to the transactional server over the mass mailing server by pro
+* **Original messages**: Several business objects have their own alias to create new records in
+  Odoo from incoming emails:
+
+  * Sales Channel (to create Leads or Opportunities in `Odoo CRM <https://www.odoo.com/page/
+    crm>`__)
+  * Support Channel (to create Tickets in `Odoo Helpdesk <https://www.odoo.com/page/helpdesk>`__)
+  * Projects (to create new Tasks in `Odoo Project <https://www.odoo.com/page
+    /project-management>`__)
+  * Job Positions (to create Applicants in `Odoo Recruitment <https://www.odoo.com/page
+    /recruitment>`__)
```
This was previous code, nice catch! 🥇

tiku-odoo commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -249,6 +185,86 @@ to give priority to the transactional server over the mass mailing server by pro
+* **Reply messages** of messages sent from Odoo are routed to their original discussion thread (and
+  to the inbox of all its followers) by the alias of the model if there is any or by the catchall
+  alias (**catchall@**). Replies to messages of models that don't have a custom alias will use the
+  catchall alias (catchall@example.odoo.com). The catchall address, however, does not have another
```
I agree- I use both because the domain will change.

tiku-odoo replied:
I've cleaned this up in both email_servers.html and email_domain.html. Thanks for pointing this out.

tiku-odoo commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -249,6 +185,86 @@ to give priority to the transactional server over the mass mailing server by pro
+  provider's settings. Leave the :guilabel:`Actions to Perform on Incoming Mails` field blank. Once
+  all the information has been filled out, click on :guilabel:`TEST & CONFIRM`.
+
+  .. image:: email_servers/incoming-server.png
```
This is the compressed version.

tiku-odoo commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -249,6 +185,86 @@ to give priority to the transactional server over the mass mailing server by pro
+be completed prior to the database going live. If a customer replies after a change is made then the
+system will not recognize the old alias and the reply won't be received.
+
+.. image:: email_servers/system-parameters.png
```
Removing this screenshot, it is large and unnecessary.

tiku-odoo commented:
@StraubCreative Thanks for your thorough review of this doc, I've made the suggested changes and made another review of the three docs (mailjet, email_domain, email_servers). They are ready for another final review/ technical review. Thanks in advance for your help on this doc.

StraubCreative commented:
@samueljlieber can you fix the build errors here please before I review.

samueljlieber commented:
Sorry for all the pushes - took me a few tries to get it right 😅

Build errors resolved in  8aaad60 . Ready for your review @StraubCreative !

StraubCreative commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -231,24 +169,102 @@ first outgoing email server (sorted by priority).
-:menuselection:`Settings --> Technical --> Outgoing` email servers. There, create two email MTA
+:menuselection:`Settings --> Technical --> Outgoing` email servers. There, create two outgoing email
 server settings; one for the transactional emails and one for the mass mailing server. Make sure
 to give priority to the transactional server over the mass mailing server by providing a lower
-priority number for the transactional MTA server.
+priority number for the transactional email server.
```
line breaks
```suggestion
server settings; one for the transactional emails and one for the mass mailing server. Make sure to
give priority to the transactional server over the mass mailing server by providing a lower priority
number for the transactional email server.
```

StraubCreative commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -231,24 +169,102 @@ first outgoing email server (sorted by priority).
+* **Reply messages** of messages sent from Odoo are routed to their original discussion thread (and
+  to the inbox of all its followers) by the alias of the model if there is any or by the catchall
+  alias (**catchall@**). Replies to messages of models that don't have a custom alias will use the
+  catchall alias (`catchall@mycompany.odoo.com`). The catchall address, however, does not have another
+  action attached to it like other aliases might, it is only used to collect replies.
```
```suggestion
  catchall alias (`catchall@mycompany.odoo.com`). The catchall address, however, does not have
  another action attached to it like other aliases might, it is only used to collect replies.
```

StraubCreative commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -231,24 +169,102 @@ first outgoing email server (sorted by priority).
+  Messages that bounce in the chatter (outside of Email Marketing) will populate a red envelope
+  indicating the failed delivery. This can be helpful to know that a Sales Order or an Invoice did
+  not reach its final destination.
+* **Original messages**: Several business objects have their own alias to create new records in
```
```suggestion
* **Original messages**: several business objects have their own alias to create new records in
```

StraubCreative commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -231,24 +169,102 @@ first outgoing email server (sorted by priority).
+* **Original messages**: Several business objects have their own alias to create new records in
+  Odoo from incoming emails:
+
+  * Sales Channel (to create Leads or Opportunities in `Odoo CRM <https://www.odoo.com/page/crm>`_)
+  * Support Channel (to create Tickets in `Odoo Helpdesk <https://www.odoo.com/page/helpdesk>`_)
+  * Projects (to create new Tasks in `Odoo Project <https://www.odoo.com/page
+    /project-management>`_)
+  * Job Positions (to create Applicants in `Odoo Recruitment <https://www.odoo.com/page
+    /recruitment>`_)
```
These are not proper nouns.
```suggestion
  * Sales channel (to create Leads or Opportunities in `Odoo CRM <https://www.odoo.com/page/crm>`_)
  * Support channel (to create Tickets in `Odoo Helpdesk <https://www.odoo.com/page/helpdesk>`_)
  * Projects (to create new Tasks in `Odoo Project <https://www.odoo.com/page
    /project-management>`_)
  * Job positions (to create Applicants in `Odoo Recruitment <https://www.odoo.com/page
    /recruitment>`_)
```

StraubCreative commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -231,24 +169,102 @@ first outgoing email server (sorted by priority).
+* Create the corresponding email addresses in the mail server (**catchall@**, **bounce@**,
+  **sales@**, etc.).
+* Set the :guilabel:`Alias Domain` name in :menuselection:`Settings --> General Settings -->
+  Discuss`. Changing the :guilabel:`Alias Domain` will change the catchall's domain for the database.
```
```suggestion
  Discuss`. Changing the :guilabel:`Alias Domain` will change the catchall's domain for the
  database.
```

StraubCreative commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -231,24 +169,102 @@ first outgoing email server (sorted by priority).
+  `catchall\@mycompany.odoo.com`).
+
+All the aliases are customizable in Odoo. Object aliases can be edited from their respective
+configuration view: :menuselection:`Settings --> Technical Menu --> Email --> Aliases`
```
```suggestion
configuration view by navigating to :menuselection:`Settings --> Technical Menu --> Email -->
Aliases`
```

StraubCreative reviewed (changes requested):
Hi @tiku-odoo 
Change requests below, thanks!

tiku-odoo commented on content/applications/general/email_communication/email_domain.rst:
```diff
@@ -1,37 +1,36 @@
-For the same reason, Odoo also recommends giving each database user an email address from the
-configured domain, rather than a generic email address domain (such as gmail.com, outlook.com,
-etc.).
+=========================================
+Configuring DNS Records for Email Sending
+=========================================
```
Noted, thanks for changing.

tiku-odoo commented on content/applications/general/email_communication/email_domain.rst:
```diff
@@ -136,16 +136,55 @@ Framework)` compliant or enable :abbr:`DKIM (DomainKeys Identified Mail)`.
 should not impact the deliverability if the :abbr:`DMARC (Domain-based Message Authentication,
 Reporting, & Conformance)` check fails.
 
+:abbr:`DMARC (Domain-based Message Authentication, Reporting, & Conformance)` records are comprised
+of tags in the form of :abbr:`DNS (Domain Name System)` records. These tags/parameters allow for
+reporting (:abbr:`RUA (Reporting URI of aggregate reports)`) & (:abbr:`RUF (Reporting URI for
+forensic reports)`) and more precise specifications like :abbr:`PCT (Percentage of messages
+subjected to filtering)`, :abbr:`P (Policy for organizational domain)`, :abbr:`SP (Policy for
+subdomains of the OD)` :abbr:`ADKIM (Alignment mode for DKIM)` & :abbr:`ASPF (Alignment mode for
+SPF)`. It should be noted that the :abbr:`DMARC (Domain-based Message Authentication, Reporting, &
+Conformance)` policy should not start out being too restrictive.
```
Thanks, I appreciate the improvements!

tiku-odoo commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -6,200 +6,138 @@ Odoo Online or Odoo.sh users
+   strongly recommended. In these Odoo hosting types email sending and receiving works instantly
+   and is monitored by professionals. Nevertheless, a company can use their own email server if
+   they want to manage the email server's reputation themselves. For more information see
+   :doc:`Configuring DNS Records for Email Sending <email_domain>`
```
Yes! Thanks 👍

tiku-odoo commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -6,200 +6,138 @@ Odoo Online or Odoo.sh users
+of the document as a notification (except to the sender). If a follower replies to the message,
+the reply updates the chatter, and Odoo relays another reply to the followers as a notification.
+Messages sent back to the chatter from users or external users will appear in the chatter from
+their respective email, or as the name listed in their Contacts record.
```
Noted, thanks.

tiku-odoo commented:
@StraubCreative 

CC: @samueljlieber 

Thanks for your edits on these two docs. The edits you suggested to the links on lines 65-66 on the email_servers.rst worked great. Thank you. 

This PR is ready for a final pass when you have a moment. 

Looking forward to see this published. 

Thanks, Tim 👍

samueljlieber commented:
Fixed build errors in https://github.com/odoo/documentation/commit/de247ea45d2fdfe831464e681c13a7960c8f43fa

@tiku-odoo please make sure you pull from PRs using `git pull --rebase` that have commits made after your commits (that should fix the pesky build errors we're getting on this PR) 🙂

StraubCreative commented on content/applications/general/email_communication/email_servers.rst:
```diff
@@ -6,200 +6,138 @@ Odoo Online or Odoo.sh users
+.. example::
+   In the following example, the from email address is replaced with the combination of the the two
+   system parameters (`mail.default.from` and `mail.catchall.domain`). This is the default\
+   notifications configuration in Odoo.
+   “Admin” <`admin@example.com`> => “Admin” <`notifications@mycompany.com`>
```
Formatting
```suggestion
   notifications configuration in Odoo: ``“Admin” <admin@example.com>`` => ``“Admin”
   <notifications@mycompany.com>``.
```

StraubCreative reviewed (approved):
Looks good @tiku-odoo .
Just a couple small things which I'll address in the following commit and pass over to DR.
Nice work 👍

StraubCreative commented:
@odoo/doc-review these updates to the "From Filter" sections in the `email_communications` documentation are ready for your review.

AntoineVDV reviewed (approved):
@robodoo r+
